cogs/timer.py

```diff
@@ -2174,6 +2174,168 @@ async def permissionCheck(self, msg, author):
+                # list of current magical items
+                charMagicList = currentItem[1]['Magic Items'].split(', ')
+                # character level
+                charLevel = int(currentItem[1]['Level'])
+            # since this checks for multiple things, this cannot be avoided
+            for u, v in startcopy.items():
+
+                for item in v:
+                    if dmChar[0] == rewardUser:
+                        break
+                    if item[0] == rewardUser:
+                        userFound = True
+                        # the player entry of the player getting the item
+                        currentItem = item
+                        # list of current consumables on the character
+                        # [1] in a player entry is the DB entry of the character
+                        charConsumableList = currentItem[1]['Consumables'].split(', ')
+                        # list of current magical items
+                        charMagicList = currentItem[1]['Magic Items'].split(', ')
+                        # character level
+                        charLevel = int(currentItem[1]['Level'])
+        
+        
+        # calculate the tier of the rewards
+        tierNum = 5
+        if charLevel < 5:
+            tierNum = 1
+        elif charLevel < 11:
+            tierNum = 2
+        elif charLevel < 17:
+            tierNum = 3
+        elif charLevel < 20:
+            tierNum = 4
+        
+        rewardCollection = db.rit
+        rewardTable = list(rewardCollection.find({"Tier": tierNum, "Minor/Major": rewardType})) #gets the full list of eligible rewards based on input
+        
+        randomItem = random.choice(rewardTable) #picks a random reward
+        rewardString = []
+        
+        rewardConsumable_group_type = "Name"
+        if "Grouped" in randomItem: # used to check for duplicates if the random reward has subtypes, like ammunition
+           rewardConsumable_group_type = "Grouped"
+        
+        player_type = "Players"
+        if rewardUser == dmChar[0]:
+            player_type = "DM"
+        
+        rewardCheck = randomItem
+        
+        # check if the item has already been rewarded to the players
+        if (rewardCheck[rewardConsumable_group_type] in dmChar[5][1][player_type]["Major"] or
+            rewardCheck[rewardConsumable_group_type] in dmChar[5][1][player_type]["Minor"]):
+            return None # returns none, which will cause the command to be repeated to try to get a non-duplicated reward item
+        
+        
+        def spellEmbedCheck(r, u):
+            sameMessage = False
+            if charEmbedmsg.id == r.message.id:
+                sameMessage = True
+            return sameMessage and ((r.emoji in alphaEmojis[:6]) or (str(r.emoji) == '❌')) and u == author
+        
+        #Puts all rewards into an array. Holdover from the out-of-timer command in reward.py
```

Comment:
Why is the other command not used? I ran a comparison of everything after with https://text-compare.com/ and they are literally the same...

Reply:
You can't import commands from another file in the cogs. Or at least, I tried to and it didn't work.

Reply:
They can, the code already does so for multiple things